Library/Homebrew/version.rb

```diff
@@ -218,6 +218,22 @@ def self.parse(spec)
     version.nil? ? NULL : new(version)
   end
 
+  COMMON_VERSION_PATTERN = /(?:(?:2\.1[12]|2\.20\.11)-|%\d{2})?((?:\d+\.)+\d+(?:[-_.~+]?(?:(?:alpha|beta|preview|pre|rc|cr|final|dev|snapshot|git|gfm)(?:\.?\d+)*|(?:ga|rel|build|[abmprv])?\.?(?:\d+\.)*\d+)|[a-f0-9]{6,}|[a-z0-9]{1,2})?)/i
```

Comment:
Sorry, I don't think this is an improvement, it's almost impossible to follow.

Reply:
That's fair, and why I put in the description comments below. Is there anything specific that makes it impossible to follow, or is it just that it's large with a bunch of alternatives?

Reply:
Could we use a `//x` regular expression, across multiple lines with inline comments, to improve readability?

Reply:
This still seems like a regression in comprehensibility even if it's split over multiple lines.

Reply:
It's just too long and will be extremely difficult to debug or modify (particularly compared to the existing code).